File: backend/ControleGastos.Api/Models/Transaction.cs
namespace ControleGastos.Api.Models;

using ControleGastos.Api.Enums;
using System.Text.Json.Serialization;

/// <summary>
/// Representa uma transação financeira vinculada a uma pessoa.
/// </summary>
public class Transaction
{
    public int Id { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Value { get; set; }
    public TransactionType Type { get; set; }
    public int PersonId { get; set; }
    [JsonIgnore]
    public Person Person { get; set; } = null!;
}

File: frontend/src/interfaces/Transaction.ts
import type { TransactionType } from "../enums/TransactionType";

export interface Transaction {
    id: number;
    description: string;
    value: number;
    type: TransactionType;
    personId: number;
}

export interface TransactionCreate {
    description: string;
    value: number;
    type: TransactionType;
    personId: number;
}